daemon/api_snapctl.go

```diff
@@ -77,14 +77,11 @@ func runSnapctl(c *Command, r *http.Request, user *auth.UserState) Response {
 				"stderr":    string(stderr),
 				"exit-code": e.ExitCode,
 			}
-			return &resp{
-				Type: ResponseTypeError,
-				Result: &errorResult{
-					Message: e.Error(),
-					Kind:    client.ErrorKindUnsuccessful,
-					Value:   result,
-				},
-				Status: 200,
+			return &apiError{
+				Status:  200,
```

Comment:
It's strange to return an error together with the `200` status. On the other hand, the code was like this before your changes too, so this should not block this PR.